crates/lib/src/transaction/validator.rs

```diff
@@ -232,61 +233,71 @@ pub async fn validate_token_payment(
     required_lamports: u64,
     signer_pubkey: Pubkey,
 ) -> Result<(), KoraError> {
-    let mut total_lamport_value = 0;
+    let mut total_lamport_value: u64 = 0;
+    let token_program = TokenKeg;
 
     for ix in transaction.message.instructions.iter() {
-        if *ix.program_id(&transaction.message.account_keys) != spl_token::id() {
+        if *ix.program_id(&transaction.message.account_keys) != token_program.program_id() {
             continue;
         }
 
-        if let Ok(spl_token::instruction::TokenInstruction::Transfer { amount }) =
-            spl_token::instruction::TokenInstruction::unpack(&ix.data)
-        {
-            let dest_pubkey = transaction.message.account_keys[ix.accounts[1] as usize];
-            let source_key = transaction.message.account_keys[ix.accounts[0] as usize];
+        // Get source account data
+        let source_key = transaction.message.account_keys[ix.accounts[0] as usize];
+        let source_account = rpc_client
+            .get_account(&source_key)
+            .await
+            .map_err(|e| KoraError::RpcError(e.to_string()))?;
 
-            let source_account = rpc_client
-                .get_account(&source_key)
-                .await
-                .map_err(|e| KoraError::RpcError(e.to_string()))?;
+        // Verify it's a token account
+        if source_account.owner != token_program.program_id() {
+            continue;
+        }
 
-            let token_account = TokenAccount::unpack(&source_account.data).map_err(|e| {
-                KoraError::InvalidTransaction(format!("Invalid token account: {}", e))
-            })?;
+        let token_state = TokenState::try_from_slice(&source_account.data)
```

Comment:
can you add a unit test for this? I'm seeing some issues when running the integration tests when this function is trying to decode the account data:
```
Validation error: Invalid mint account data: An account's data contents was invalid
```